### PR TITLE
Implement HEEx tokenizer and a transformer to convert it to JSON

### PR DESCRIPTION
**Pull request by cibernox**
Closes https://github.com/BeaconCMS/beacon-ui-builder/issues/13

### CODE REVIEW

File: lib/beacon/template/json_transformer.ex [hunk 1]
```diff
@@ -0,0 +1,128 @@
+defmodule Beacon.Template.HEEx.JsonTransformer do
+  # Transforms the AST returned by Beacon.Template.HEEx.Tokenizer.parse into a
```
Comment (leandrocp) [inline lib/beacon/template/json_transformer.ex:2]: You can transform this comment block into a proper `@moduledoc`

File: lib/beacon_web/controllers/api/page_json.ex [hunk 2]
```diff
@@ -18,14 +19,19 @@ defmodule BeaconWeb.API.PageJSON do
   end
 
   defp data(%Page{} = page) do
+    Logger.debug("page.template")
+    Logger.debug(page.template)
```
Comment (leandrocp) [inline lib/beacon_web/controllers/api/page_json.ex:23]:
```suggestion
```

File: lib/beacon_web/controllers/api/page_json.ex [hunk 3]
```diff
@@ -2,6 +2,7 @@ defmodule BeaconWeb.API.PageJSON do
   @moduledoc false
 
   alias Beacon.Content.Page
+  require Logger
```
Comment (leandrocp) [inline lib/beacon_web/controllers/api/page_json.ex:5]:
```suggestion
```

File: priv/static/beacon_admin.css
Comment (leandrocp) [inline priv/static/beacon_admin.css:1]: Please revert this file, I'll fix this issue https://github.com/BeaconCMS/beacon/issues/317 asap

File: priv/static/beacon_admin.js
Comment (leandrocp) [inline priv/static/beacon_admin.js:1]: Please revert this file, I'll fix this issue https://github.com/BeaconCMS/beacon/issues/317 asap

File: test/beacon/template/json_transformer_test.exs [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                   "\n  ",
+                   %{attrs: %{}, content: [%{attrs: %{}, content: "user.name", tag: :eex}], tag: "p"},
+                   "\n  ",
+                   %{
+                     arg: "if true do",
+                     blocks: [
+                       %{content: [" ", %{attrs: %{}, content: ["this"], tag: "p"}], key: "else"},
+                       %{content: [%{attrs: %{}, content: ["that"], tag: "p"}], key: "end"}
+                     ],
+                     tag: :eex_block
```
Comment (leandrocp) [inline test/beacon/template/json_transformer_test.exs:63]: Maybe convert this atom to string? To define that `tag` is always a string to keep the same pattern everywhere?

**Review by leandrocp — CHANGES_REQUESTED** [review]
Hey @cibernox good one, left just a few comments.

File: test/beacon/template/json_transformer_test.exs (hunk 4 above)
Comment (cibernox) [reply test/beacon/template/json_transformer_test.exs]: Done

File: lib/beacon/template/json_transformer.ex (hunk 1 above)
Comment (cibernox) [reply lib/beacon/template/json_transformer.ex]: done

**Review by leandrocp — APPROVED** [review]
🎉

**Pull request merged**